sphinx/ext/doctest.py

```diff
@@ -107,41 +106,52 @@ def run(self):
+                        "missing '+' or '-' in '%s' option." % option,
+                        line=self.lineno)
                     continue
                 flag = doctest.OPTIONFLAGS_BY_NAME[option[1:]]  # type: ignore
                 node['options'][flag] = (option[0] == '+')
         if self.name == 'doctest' and 'pyversion' in self.options:
             try:
                 option = self.options['pyversion']
                 option_strings = option.split()
-                # :pyversion: >= 3.6   -->   op='>=', version='3.6'
+                # :pyversion: >= 3.6   -->   operand='>=', version='3.6'
                 operand, version = [item.strip() for item in option_strings]
-                operands = ('<=', '<', '==', '>=', '>')
-                if operand not in operands:
-                    self.state.document.reporter.warning(
-                            "'%s' is not a valid pyversion operand.\n" 
-                            "Avaliable operands: %s" % (operand, operands), 
-                            line=self.lineno)
-                else:
-                    rv = V(platform.python_version())  # Running version
-                    sv = V(version)  # Specified version
-                    skip = ((operand == '<=' and not (rv <= sv)) or
-                            (operand == '<'  and not (rv < sv)) or
-                            (operand == '==' and not (rv == sv)) or
-                            (operand == '>=' and not (rv >= sv)) or
-                            (operand == '>'  and not (rv > sv)))
-                    if skip:
-                        flag = doctest.OPTIONFLAGS_BY_NAME['SKIP']
-                        node['options'][flag] = True
+                if not self.proper_pyversion(operand, version):
+                    flag = doctest.OPTIONFLAGS_BY_NAME['SKIP']
+                    node['options'][flag] = True  # Skip the test
             except ValueError:
                 self.state.document.reporter.warning(
-                        "'%s' is not a valid pyversion value" % option, 
-                        line=self.lineno)
+                    "'%s' is not a valid pyversion value" % option,
+                    line=self.lineno)
 
         return [node]
 
+    def proper_pyversion(self, operand, version):
+        """Compare `version` to the Python version, relying on `operand`.
+
+        This function is meant to be used to evaluate the doctest :pyversion:
+        option. For instance, if the doctest directive provides the option
+        :pyversion: >= 3.3, then we have to check if the running Python version
+        is greather or equal to 3.3. In that case, operand will be the string
+        '>=' and version the string '3.3'. proper_pyversion() will return True
+        if the running Python version is >= 3.3, False otherwise.
+        """
+        operands = ('<=', '<', '==', '>=', '>')
+        if operand not in operands:
+            self.document.reporter.warning(
```

Comment:
Your interface is just perfect :-)  So, I will change my interface following your suggestion, fix the typo and check (with pytest) everything works properly. Thank you very much!